src/Diagnostics.DataProviders/ChangeAnalysisClient.cs

```diff
@@ -0,0 +1,144 @@
+            List<ChangeSetResponseModel> changeSetsResponse = JsonConvert.DeserializeObject<List<ChangeSetResponseModel>>(jsonString);
+            return changeSetsResponse;
+        }
+
+        /// <inheritdoc/>
+        public async Task<ResourceIdResponseModel> GetResourceIdAsync(List<string> hostnames, string subscription)
+        {
+            string requestUri = changeAnalysisEndPoint + "resourceId?api-version=2019-04-01-preview";
+            object requestBody = new
+            {
+                hostNames = hostnames,
+                subscriptionId = subscription
+            };
+            string jsonString = await PrepareAndSendRequest(requestUri, requestBody);
+            if (!string.IsNullOrWhiteSpace(jsonString))
+            {
+                ResourceIdResponseModel resourceIdResponse = JsonConvert.DeserializeObject<ResourceIdResponseModel>(jsonString);
+                return resourceIdResponse;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Prepares httpwebrequest to <paramref name="requestUri"/> with <paramref name="postBody"/> as body of the request.
+        /// </summary>
+        /// <param name="requestUri">Change Analysis Request URI</param>
+        /// <param name="postBody">Body of the request</param>
+        /// <returns>JSON string received from <paramref name="requestUri"/>.</returns>
+        private async Task<string> PrepareAndSendRequest(string requestUri, object postBody)
+        {
+            HttpRequestMessage requestMessage = new HttpRequestMessage(HttpMethod.Post, requestUri);
+            string authToken = await ChangeAnalysisTokenService.Instance.GetAuthorizationTokenAsync();
+
+            // Add required headers.
+            requestMessage.Headers.Add("Authorization", authToken);
+            requestMessage.Headers.Add("x-ms-client-object-id", clientObjectIdHeader);
+
+            // For requests coming from Diagnose and Solve, add x-ms-principal-name header.
+            if (!string.IsNullOrWhiteSpace(clientPrincipalNameHeader))
+            {
+               requestMessage.Headers.Add("x-ms-principal-name", clientPrincipalNameHeader);
+            }
+
+            requestMessage.Content = new StringContent(JsonConvert.SerializeObject(postBody), Encoding.UTF8, "application/json");
+            CancellationTokenSource cancellationTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(DataProviderConstants.DefaultTimeoutInSeconds));
+            try
+            {
+                HttpResponseMessage responseMessage = await httpClient.SendAsync(requestMessage, cancellationTokenSource.Token);
+                string content = await responseMessage.Content.ReadAsStringAsync();
+                if (!responseMessage.IsSuccessStatusCode)
+                {
+                    throw new Exception(content);
+                }
+
+                return content;
+            }
+            catch (Exception ex)
```

Comment:
I don't think you need a try..catch.. statement here since you are not doing anything with the exception

Reply:
You are right. I think I should be logging the request in Kusto instead of doing nothing with it.

Reply:
There is logging already happening at `DataProviderLogDecorator`. Unless you want to add any special logging here

Reply:
I just saw what the `DataProviderLogDecorator` does. I was not aware of this class before. I will add it